polar-core/src/vm.rs

```diff
@@ -1452,20 +1452,7 @@ impl PolarVirtualMachine {
                 let right = args.pop().unwrap();
                 let left = args.pop().unwrap();
                 match (left.value(), right.value()) {
-                    (Value::Variable(var), _) => match self.variable_state(var) {
-                        VariableState::Unbound => {
-                            self.push_goal(Goal::Unify { left, right })?;
-                        }
-                        _ => {
-                            return Err(self.type_error(
-                                &left,
-                                format!(
-                                    "Can only assign to unbound variables, {} is not unbound.",
-                                    var.to_polar()
-                                ),
-                            ));
-                        }
-                    },
+                    (Value::Variable(var), _) => self.bind(var, right)?,
```

Comment:
If the intention is to fail on bound values, we need to first check the variable state. If `right` is an unbound variable and `var` is bound `bind` will bind `right` to the value of `var`.

With two variable arguments `bind` binds them together and ensures they have the same value, it is bi-directional.

(This part of the interface was a bit up for debate, so if it's counterintuitive we could change that)

Reply:
oh, interesting. Okay